packages/database/src/TupaiaDatabase.js

```diff
@@ -45,7 +45,7 @@ export const JOIN_TYPES = {
 };
 
 // list valid behaviour so we can validate against sql injection
-const VALID_CAST_TYPES = ['text', 'text[]', 'date'];
+const VALID_CAST_TYPES = ['text', 'text[]', 'date', 'entity_type[]'];
```

Comment:
Maybe 'enum' is the cast type so it can stay generic, and specify an enum type `castAsEnum: 'entity_type'`?

Reply:
N/m, removed in later commit